### PR TITLE
Fix Column Reference

### PR DESCRIPTION
**Pull request by MattsonCam**
Fixed bug by using correct columns to check if values are the same between corresponding group columns, and also incremented the version patch.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "pairwise-compare"
-version = "0.1.1"
+version = "0.1.2"
```
Comment (d33bs) [inline pyproject.toml:3]: Consider using [`poetry-dynamic-versioning`](https://github.com/mtkennerly/poetry-dynamic-versioning) to instead rely on git tags for versions. If you tie this to GitHub releases you can also use GH Actions to automatically deploy your package to PyPI by way of [trusted publishing](https://docs.pypi.org/trusted-publishers/). The advantage becomes that you aren't tied to making a pull request to modify the version for new releases.

File: src/comparison_tools/PairwiseCompare.py
Comment (d33bs) [inline src/comparison_tools/PairwiseCompare.py:1]: The changes here are specific enough that it made me wonder if a test case should be added somewhere. Specifically, could an edge case be added which originally triggers the bug (even using static examples)? This might help assure that the package can't encounter the bug again with future changes.

File: src/comparison_tools/PairwiseCompare.py [hunk 2]
```diff
@@ -145,11 +145,11 @@ def __get_group_column_element(
         else:
             return _group_column_data
 
-    def __contains_match(self, _groups):
+    def __contains_match(self, _groups, _group_columns):
```
Comment (d33bs) [inline src/comparison_tools/PairwiseCompare.py:148]: Consider adding type hints to this definition. These can be added gradually without impacting other areas of the code.

File: src/comparison_tools/PairwiseCompare.py [hunk 3]
```diff
@@ -212,7 +212,7 @@ def inter_comparisons(self):
             # Iterate through each well group cartesian product and save the data
             for ppair in comparison_key_product:
 
-                if self.__contains_match(ppair):
+                if self.__contains_match(ppair, self.__posthoc_group_cols):
```
Comment (d33bs) [inline src/comparison_tools/PairwiseCompare.py:215]: Consider adding comments here and throughout to help detail these if-scenarios and what this does.

**Review by d33bs — APPROVED** [review]

File: src/comparison_tools/PairwiseCompare.py (hunk 2 above)
Comment (MattsonCam) [reply src/comparison_tools/PairwiseCompare.py:148]: Good idea, surprised nobody mentioned this before. I'll change this

File: src/comparison_tools/PairwiseCompare.py
Comment (MattsonCam) [reply src/comparison_tools/PairwiseCompare.py:1]: I've thought about this too. So far I've been creating tests directly for the pairwise compare manager, but I think I can still cover this bug partially by testing the pairwise compare manager, so I'll change a test case

File: pyproject.toml [hunk 4]
```diff
@@ -28,6 +28,12 @@ matplotlib = "^3.9.2"
 pylint = "^3.3.1"
 notebook = "^7.2.2"
 
+[tool.poetry.requires-plugins]
+poetry-dynamic-versioning = { version = ">=1.0.0,<2.0.0", extras = ["plugin"] }
+
+[tool.poetry-dynamic-versioning]
+enable = true
```
Comment (d33bs) [inline pyproject.toml:35]: Consider specifying further what the style and version control types are here. 

```suggestion
enable = true
style = "pep440"
vcs = "git"
```

File: pyproject.toml [hunk 5]
```diff
@@ -28,6 +28,12 @@ matplotlib = "^3.9.2"
...
+
```
Comment (d33bs) [inline pyproject.toml:36]: Consider adding a substitution area for PDV where it can supply a version number for reference within the package itself. This is useful for quickly checking which version of a package is installed; e.g. `comparison_tools.__version__` from an addition in the `__init__.py`: `__version__ = "0.0.0"`

```suggestion

[tool.poetry-dynamic-versioning.substitution]
files = [ "src/comparison_tools/__init__.py" ]
```

**Review by d33bs — APPROVED** [review]
With a move to PDV you may want to consider tagging the previous commits with the related git tag for the version reference at some point. This won't impact things moving forward but could help reduce confusion in the future.

File: pyproject.toml (hunk 4 above)
Comment (MattsonCam) [reply pyproject.toml:35]: Added!

File: pyproject.toml (hunk 5 above)
Comment (MattsonCam) [reply pyproject.toml:36]: Added!

**Comment by MattsonCam** [comment]
Thanks for the suggestions @d33bs, merging now!
